backend/src/sync.js

```diff
@@ -149,6 +151,59 @@ async function saveBlocks(blocksInfo) {
   }
 }
 
+async function saveGenesis(time, records) {
+  try {
+    await models.sequelize.transaction(async transaction => {
+      try {
+        await Promise.all([
+          models.AccessKey.bulkCreate(
+            records
+              .filter(record => record.AccessKey !== undefined)
+              .map(record => {
+                let accessKeyType;
+                const permission = record.AccessKey.access_key.permission;
+                if (typeof permission === "string") {
+                  accessKeyType = permission;
+                } else if (permission !== undefined) {
+                  accessKeyType = Object.keys(permission)[0];
+                } else {
+                  throw new Error(
+                    `Unexpected error during access key permission parsing in transaction ${
+                      tx.hash
+                    }: 
+                    the permission type is expected to be a string or an object with a single key, 
+                    but '${JSON.stringify(permission)}' found.`
+                  );
+                }
+                return {
+                  accountId: record.AccessKey.account_id,
+                  publicKey: record.AccessKey.public_key,
+                  accessKeyType
+                };
+              })
+          ),
+          models.Account.bulkCreate(
+            records
+              .filter(record => record.Account !== undefined)
+              .map((record, index) => {
+                return {
+                  accountId: record.Account.account_id,
+                  accountIndex: time + index,
```

Comment:
You use `timestamp * 1000000 + index` in another place. Are they synced?

Reply:
fixed